dev-docs/source/ReleaseChecklist.rst

```diff
@@ -110,8 +110,8 @@ with the Release Manager.
 Monday, 3 weeks
 ---------------
 
-*  Ensure that Manual testing begins. An overview of what Manual testing is can be found
-   `here <https://www.mantidproject.org/Unscripted_Manual_Testing>`__. Generate the
+*  Ensure that Manual testing begins. An overview of the Manual testing to be done is
+   found `here <https://developer.mantidproject.org/Testing/index.html>`__. Generate the
```

Comment:
Thanks. Can we use an internal reference to avoid being domain specific:
\`here \<Testing\>\`

The link is defined at https://raw.githubusercontent.com/mantidproject/mantid/master/dev-docs/source/Testing/index.rst